### PR TITLE
Document ability to add custom filesystems (Eg. Dropbox)

### PR DESCRIPTION
**Pull request by brenelz**

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Fix the cs and squash to one commit.

File: filesystem.md [hunk 1]
```diff
@@ -106,3 +107,35 @@ The `Storage` facade may be used to interact with any of your configured disks.
 #### Delete A Directory
 
 	Storage::deleteDirectory($directory);
+
+<a name="custom-filesystems"></a>
+## Custom Filesystems
+In order to set up your custom filesystem you'll need to create a Service Provider (For example, `DropboxFilesystemServiceProvider`. In its `boot` method, you'll want to inject an instance of the `Illuminate\Contracts\Filesystem\Factory` contract, and then call the `extend` method of the injected instance.
```
Comment (mattstauffer) [inline filesystem.md]: Missing the closing paren on the "For example"; also, it should be "for example", no capital.

File: filesystem.md [hunk 2]
```diff
@@ -106,3 +107,35 @@ The `Storage` facade may be used to interact with any of your configured disks.
 #### Delete A Directory
 
 	Storage::deleteDirectory($directory);
+
+<a name="custom-filesystems"></a>
+## Custom Filesystems
```
Comment (mattstauffer) [inline filesystem.md]: Could use an intro paragraph. Like:

> Laravel's Flysystem integration provides for local filesystem, Amazon S3, and Rackspace Cloud Storage. But Flysystem has adapters for many additional storage systems. If you want to use one of these additional adapters in your Laravel application, you'll want to create a Custom Filesystem.

Or something. :)

File: filesystem.md [hunk 3]
```diff
@@ -106,3 +107,35 @@ The `Storage` facade may be used to interact with any of your configured disks.
...
+
+	<?php namespace App\Providers;
+
+	use Illuminate\Contracts\Filesystem\Factory as FilesystemFactory;
+
+	class DropboxFilesystemServiceProvider {
+		
+		public function boot(FilesystemFactory $filesystem)
+		{
+			$filesystem->extend('dropbox',function($app, $config)
```
Comment (mattstauffer) [inline filesystem.md]: space after comma:
```php
$filesystem->extend('dropbox', function($app, $config)
```

File: filesystem.md [hunk 4]
```diff
@@ -106,3 +107,35 @@ The `Storage` facade may be used to interact with any of your configured disks.
...
+
+	use Illuminate\Contracts\Filesystem\Factory as FilesystemFactory;
+
+	class DropboxFilesystemServiceProvider {
+		
+		public function boot(FilesystemFactory $filesystem)
+		{
+			$filesystem->extend('dropbox',function($app, $config) 
+			{
+				$client = new Client($config['accessToken'],$config['clientIdentifier']);
```
Comment (mattstauffer) [inline filesystem.md]: space after comma

**Comment by barryvdh** [comment]
Why did you close this?

**Comment by brenelz** [comment]
Fully revised pull request here - https://github.com/laravel/docs/pull/1140

**Pull request closed without merging**